integration_tests/src/main/python/conditionals_test.py

```diff
@@ -379,3 +379,30 @@ def test_case_when_all_then_values_are_scalars_with_nulls():
         "tab",
         sql_without_else,
         conf = {'spark.rapids.sql.case_when.fuse': 'true'})
+
+@pytest.mark.parametrize('combine_string_contains_enabled', ['true', 'false'])
+def test_combine_string_contains_in_case_when(combine_string_contains_enabled):
+    data_gen = [("c1", string_gen)]
+    sql =  """
+            SELECT
+                CASE
+                     WHEN INSTR(c1, 'a') > 0 THEN 'a'
+                     WHEN INSTR(c1, 'b') > 0 THEN 'b'
+                     WHEN INSTR(c1, 'c') > 0 THEN 'c'
+                     ELSE ''
+                END as output_1,
+                CASE
+                     WHEN INSTR(c1, 'c') > 0 THEN 'c'
+                     WHEN INSTR(c1, 'd') > 0 THEN 'd'
+                     WHEN INSTR(c1, 'e') > 0 THEN 'e'
+                     ELSE ''
+                END as output_2
+            from tab
+            """
+    # spark.rapids.sql.combined.expressions.enabled is true by default
+    assert_gpu_and_cpu_are_equal_sql(
```

Comment:
How do we know this test will run into the new `GpuMultiContains`?

Reply:
We can trust this config: `spark.rapids.sql.expression.combined.GpuContains`.
I'm sure this config takes effective by adding logs to verify.
Of course, I did not commit the `log print` code changes.

Reply:
Maybe we can try `assert_cpu_and_gpu_are_equal_sql_with_capture` with specifying the `GpuMultiContains` class name ?

Reply:
I tried, but failed. It's an internal GPU plan transformation, we can not detect this internal plan transformation.

Reply:
Not sure if we need to figure out a way to do this.